src/domain/Product.ts

```diff
@@ -0,0 +1,25 @@
+interface IProduct {
+  name: string;
+  price: number;
+  quantity: number;
+}
+
+class Product implements IProduct {
+  id: string;
+  name: string;
+  price: number;
+  quantity: number;
+
+  constructor(product: Product, id = Math.random().toString(36).substring(2, 9)) {
+    this.id = id;
+    this.update(product);
+  }
+
+  update(product: Product) {
+    this.name = product.name;
+    this.price = product.price;
+    this.quantity = product.quantity;
+  }
```

Comment:
```suggestion
  constructor(product: Product, id = Math.random().toString(36).substring(2, 9)) {
    this.validatePrice(product.price);
    this.validateQuantity(product.quantity);

    this.id = id;
    this.update(product);
  }

  update(product: Product) {
    this.validatePrice(product.price);
    this.validateQuantity(product.quantity);

    this.name = product.name;
    this.price = product.price;
    this.quantity = product.quantity;
  }

  private validatePrice(price: number) {
    if (price % CONFIGURATION.AMOUNT.UNIT !== 0) {
      throw new Error(ERROR_MESSAGE.INCORRECT_UNIT_PRODUCT_PRICE);
    }
  }

  private validateQuantity(price: number) {
    if (price > CONFIGURATION.QUANTITY.MAX) {
      throw new Error(ERROR_MESSAGE.INCORRECT_UNIT_PRODUCT_QUANTITY);  // required속성으로 필터링을해서 필요성에 의구심이 있긴하지만 예시를 위해 추가해봤어요 👀 
    }
  }
```
이렇게 도메인 내부에서 도메인 내부 데이터를 검증하면 좋을 것 같아 피드백을 드렸었어요!

Reply:
아 이런 방식이었군요! 😯